src/frontend/mod.rs

```diff
@@ -583,56 +582,42 @@ impl MainPodBuilder {
         Ok(st)
     }
 
+    /// For every operation that has Entry statements as arguments we add a Contains statement to
+    /// open the dictionary.
+    fn add_entries_contains(&mut self, op: &Operation) -> Result<()> {
+        for arg in &op.1 {
+            if let OperationArg::Statement(Statement::Contains(
+                ValueRef::Literal(dict),
+                ValueRef::Literal(key),
+                ValueRef::Literal(v),
+            )) = arg
+            {
+                let root_key = (dict.clone(), key.clone());
+                if !self.dict_contains.contains(&root_key) {
```

Comment:
What happens if:
1. I add a `DictContains` or `Contains` operation
2. I use the resulting `Contains` statement as an argument to another operation

It seems that `self.dict_contains` doesn't know about the operation in (1), and so we would add a duplicate operation here.

Reply:
Yes, your observation is correct!  I wrote a TODO here about this:
https://github.com/0xPARC/pod2/blob/b11dd9317bb623467ce95c9d8d06292491bb83b6/src/frontend/mod.rs#L131

I'll create an issue for this as I prefer addressing it in a future PR

Reply:
https://github.com/0xPARC/pod2/issues/400